sycl/include/sycl/queue.hpp

```diff
@@ -2757,7 +2758,8 @@ class __SYCL_EXPORT queue : public detail::OwnerLessBase<queue> {
     if constexpr (!(detail::KernelLambdaHasKernelHandlerArgT<KernelType,
                                                              void>::value)) {
       return detail::submit_kernel_direct_single_task<KernelName, true>(
-          *this, KernelFunc, {}, Properties, TlsCodeLocCapture.query());
+          *this, KernelFunc, {}, std::forward<PropertiesT>(Properties),
```

Comment:
Why haven't you updated the else path? And what about the subsequent calls down to the stack? If you forward here, but all next calls in the stack still accept by value, then this optimization makes little sense.